### ISSUE TITLE
Full Hessian for CNN/BatchNorm -- Feature Request 

### ISSUE THREAD

**Issue by opooladz** [cybertronai/pytorch-sso#2]
Hello Kazuki, 

I am wondering where the implementation limitations arise for the full hessian method for CNN/BatchNorm? I understand that using the full hessian and calculating the inverse is costly in time and space but I still want to run some experiments using a gauss newton approximation of the hessian and possibly some faster inversion methods. Is it possible to get the full hessian method implementation working? Can I get some direction?

Thanks in advance
